fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1341,42 +1345,133 @@ class GroupSpec {
+            }
+            recordQuery = RecordQuery.newBuilder()
+                    .setRecordType(COMPLEX_DOC)
+                    .setFilter(filter)
+                    .setSort(field(partitionFieldName), sortType != SortType.ASCENDING)
+                    .build();
         }
 
-        final List<LuceneQueryClause> luceneQueryClauses;
+        LucenePlanner planner = new LucenePlanner(recordStore.getRecordMetaData(), recordStore.getRecordStoreState(), PlannableIndexTypes.DEFAULT, recordStore.getTimer());
+        RecordQueryPlan plan = planner.plan(recordQuery);
+        assertTrue(plan instanceof LuceneIndexQueryPlan);
+        LuceneIndexQueryPlan luceneIndexQueryPlan = (LuceneIndexQueryPlan) plan;
+        LuceneScanParameters scanParameters = (LuceneScanParameters)luceneIndexQueryPlan.getScanParameters();
+        return (LuceneScanQuery)scanParameters.bind(recordStore, index, EvaluationContext.forBinding("group_value", 1));
+    }
 
-        if (comparisonType == Type.NOT_EQUALS) {
-            luceneQueryClauses = List.of(new LuceneQueryMultiFieldSearchClause(LuceneQueryType.QUERY, luceneSearch, false));
-        } else {
-            luceneQueryClauses = List.of(new LuceneQueryMultiFieldSearchClause(LuceneQueryType.QUERY, luceneSearch, false),
-                    new LuceneQueryFieldComparisonClause.LongQuery(
-                            LuceneQueryType.QUERY, isSynthetic ? "complex_timestamp" : "timestamp",
-                            LuceneIndexExpressions.DocumentFieldType.LONG,
-                            new Comparisons.SimpleComparison(comparisonType, predicateComparand),
-                            false,
-                            null));
+    @ParameterizedTest
+    @ValueSource(booleans = { true, false })
+    void partitionFieldPredicateDetectionTest(boolean isSynthetic) {
+        final Map<String, String> options = Map.of(
+                INDEX_PARTITION_BY_FIELD_NAME, isSynthetic ? "complex.timestamp" : "timestamp",
+                INDEX_PARTITION_HIGH_WATERMARK, String.valueOf(8));
+        Pair<Index, Consumer<FDBRecordContext>> indexConsumerPair = setupIndex(options, true, isSynthetic);
+        final Index index = indexConsumerPair.getLeft();
+        Consumer<FDBRecordContext> schemaSetup = indexConsumerPair.getRight();
+
+        final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_REPARTITION_DOCUMENT_COUNT, 8)
+                .build();
+        final String luceneSearch = isSynthetic ? "simple_text:about" : "text:about";
+
+        try (FDBRecordContext context = openContext(contextProps)) {
+            schemaSetup.accept(context);
+            LucenePartitioner partitioner = getIndexMaintainer(index).getPartitioner();
+
+            for (Comparisons.Type comparisonType : List.of(Type.NOT_EQUALS, Type.LESS_THAN, Type.LESS_THAN_OR_EQUALS, Type.GREATER_THAN, Type.LESS_THAN_OR_EQUALS, Type.EQUALS)) {
```

Comment:
This is good, but it seems like we should have some more negative tests, such as using `or`, or comparing a different field, or putting it in the lucene string.
(probably if it is in the lucene string we should comment that parsing the string and supporting this is probably possible, but not a priority right now.